plugins/dev-tools/src/toolboxCategories.js

```diff
@@ -1,340 +1,439 @@
+          'type': 'controls_if',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_compare',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_operation',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_negate',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_boolean',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_null',
+        },
+        {
+          'kind': 'block',
+          'type': 'logic_ternary',
+        },
+      ]
+    },
+    {
+      'kind': 'category',
+      'name': 'Loops',
+      'contents': [
+        {
+          'kind': 'block',
+          "type": "controls_repeat_ext",
+          "inputs": {
+            "TIMES": {
+              "shadow": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 10
+                }
+              }
+            }
+          }
+        },
+        {
+          'kind': 'block',
+          'type': 'controls_repeat',
+          'disabled': 'true'
+        },
+        {
+          'kind': 'block',
+          'type': 'controls_whileUntil',
+        },
+        {
+          "kind": "block",
+          "type": "controls_for",
+          "inputs": {
+            "FROM": {
+              "block": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 1
+                }
+              }
+            },
+            "TO": {
+              "block": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 10
+                }
+              }
+            },
+            "BY": {
+              "block": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 1
+                }
+              }
+            },
+          }
+        },
+        {
+          'kind': 'block',
+          'type': 'controls_forEach',
+        },
+        {
+          'kind': 'block',
+          'type': 'controls_flow_statements',
+        },
+      ]
+    },
+    {
+      'kind': 'category',
+      'name': 'Math',
+      'contents': [
+        {
+          'kind': 'block',
+          'type': 'math_number',
+          'fields': {
+            'NUM': 123
+          }
+        },
+        {
+          "kind": "block",
+          "type": "math_arithmetic",
+          "inputs": {
+            "A": {
+              "shadow": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 1
+                }
+              }
+            },
+            "B": {
+              "shadow": {
+                "type": "math_number",
+                "fields": {
+                  "NUM": 1
+                }
+              }
+            }
+          }
+        },
+        {
+          "kind": "block",
+          "type": "math_single",
+          "inputs": {
+            //unsure what goes here.
```

Comment:
Here's the JSON generated using the [playground](https://blockly-demo.appspot.com/static/tests/playground.html)!

```
{
  "blocks": {
    "languageVersion": 0,
    "blocks": [
      {
        "type": "math_single",
        "id": "7P#.3msd$=xW5`G9UN2q",
        "x": 212,
        "y": 212,
        "fields": {
          "OP": "ROOT"
        },
        "inputs": {
          "NUM": {
            "shadow": {
              "type": "math_number",
              "id": "MEt,JJH^y`bZ#2{9C$L}",
              "fields": {
                "NUM": 9
              }
            }
          }
        }
      }
    ]
  }
}
```

You def want to clean it up a bit (e.g. removing IDs) but hopefully that shows you how it works =)

Best wishes! --Beka